server/src/main/java/com/cloud/storage/VolumeApiServiceImpl.java

```diff
@@ -2396,6 +2397,35 @@ private void checkDeviceId(Long deviceId, VolumeInfo volumeToAttach, UserVmVO vm
         return volumeToAttach;
     }
 
+    protected void validateIfVmHasBackups(UserVmVO vm, boolean attach) {
+        if ((vm.getBackupOfferingId() != null || CollectionUtils.isNotEmpty(vm.getBackupVolumeList())) &&
```

Comment:
We could invert this `if` to reduce indentation.

Reply:
Done, thanks.